core/src/main/java/org/sonarsource/sonarlint/core/websocket/WebSocketService.java

```diff
@@ -243,6 +255,14 @@ private void removeProjectsFromSubscriptionListForConnection(String updatedConne
     }
   }
 
+  private void handleEventDispatcher(String connectionId) {
+    this.eventRouterByConnectionId.put(connectionId,
+      new EventDispatcher()
+        .dispatch(QualityGateChangedEvent.class, new ShowSmartNotificationOnQualityGateChangedEvent(client, configurationRepository, telemetryService))
+        .dispatch(IssueChangedEvent.class, new UpdateStorageOnIssueChanged(storageFacade.connection(connectionId)))
```

Comment:
As we have the same handlers for SQ (except `QualityGateChangedEvent`), here you could probably mix this dispatcher with the one configured for SQ. There is probably a way to avoid the double declaration

Reply:
Thinking about it a bit more, maybe it's better to keep 2 separate dispatchers for now. Even if there is some redundancy/duplication, we don't support exactly the same events.